src/test/groovy/graphql/normalized/ExecutableNormalizedOperationToAstCompilerTest.groovy

```diff
@@ -1370,6 +1356,34 @@ class ExecutableNormalizedOperationToAstCompilerTest extends Specification {
       }
     }
   }
+}
+'''
+    }
+
+    def "introspection query can be printed __type"() {
+        def sdl = '''
+        type Query {
+            f: String 
+        }
+        '''
+        def query = '''
+        query introspection_query {
+            __type(name: "World") {
+                name
+                fields {
+                    name
+                }
+            }
+        }
+        '''
+
+        GraphQLSchema schema = mkSchema(sdl)
+        def fields = createNormalizedFields(schema, query)
+        when:
+        def result = compileToDocument(schema, QUERY, null, fields, noVariables)
```

Comment:
This line is different to latest master because v21 does not have any defer changes

Master branch has the method `localCompileToDocument`